src/components/Task/SimpleTaskInterface.tsx

```diff
@@ -0,0 +1,86 @@
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import { noneString } from 'common/constants';
+import { DetailsGroup } from 'components/common';
+import { useCommonStyles } from 'components/common/styles';
+import {
+    formatType,
+    getInputDefintionForLiteralType
+} from 'components/Launch/LaunchWorkflowForm/utils';
+import { Task, Variable } from 'models';
+import * as React from 'react';
+
+const useStyles = makeStyles((theme: Theme) => ({
+    label: {
+        marginRight: theme.spacing(1)
+    },
+    typeAnnotationContainer: {
+        paddingLeft: theme.spacing(0.5)
+    },
+    typeAnnotation: {
+        color: theme.palette.secondary.main
+    }
+}));
+
+const emptyVariables = {
+    variables: {}
+};
+
+const VariablesList: React.FC<{ variables: Record<string, Variable> }> = ({
+    variables
+}) => {
+    const commonStyles = useCommonStyles();
+    const styles = useStyles();
+    const output = Object.keys(variables).reduce<React.ReactNode[]>(
```

Comment:
Do you have a little test for this kind of reducer? I'm sure it works if you've got it running; for whatever reason the combination of the calls to `push` and returning the object feels counterintuitive to me.

Reply:
I don't. But since this is code that is internal to the component, I think the right test would be ensuring that the component renders the correct strings for a few given combinations of values? I'll note it down and do a follow-up to add it later. Thank you!